compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/ComposeSceneLayer.skiko.kt

```diff
@@ -190,3 +192,18 @@ internal fun rememberComposeSceneLayer(
     }
     return layer
 }
+
+
+/**
+ * Sets the content of the layer to [content].
+ */
+@Composable
+internal fun ComposeSceneLayer.applyContent(content: @Composable () -> Unit) {
+    val currentContent by rememberUpdatedState(content)
+    DisposableEffect(this) {
+        setContent {
+            currentContent()
+        }
```

Comment:
No need to create one more lambda
```suggestion
        setContent(currentContent)
```

Reply:
I think there is, actually.

If we do 
```
    val currentContent by rememberUpdatedState(content)
    DisposableEffect(this) {
        setContent(currentContent)
        onDispose {  }
    }
```
it won't cause the effect to re-run when the value of `currentContent` changes.

So we'll need to add `currentContent` to the keys of `DisposableEffect`, which negates the point of wrapping `content` in `rememberUpdatedState`. It will cause a call to `ComposeSceneLayer.setContent` when `currentContent` changes, which recreates the composition, which is unnecessary here.

The current version, 
```
    val currentContent by rememberUpdatedState(content)
    DisposableEffect(this) {
        setContent {
            currentContent()
        }
        onDispose {  }
    }
```
passes the same lambda to `setContent` every time, as it doesn't capture anything that changes (`currentContent` is a delegate that doesn't change; only the value it returns changes). When the value returned by `currentContent` changes, this is just a regular recomposition inside the child Composition; `setContent` itself isn't called again.

Reply:
Actually, following this discussion: https://kotlinlang.slack.com/archives/G010KHY484C/p1713274994776799 there doesn't seem to be a need to rememberUpdatedState the content here; doing it only in the (two) callers is enough.

Reply:
It seems we still need it according the last comment. `content` lambda may change.

Reply:
> It seems we still need it according the last comment. content lambda may change.

Not in the cases we have right now, and I'm not sure it's worth "protecting" against unlikely future uses. Especially since nothing terrible will happen even if we don't (worst case - `setContent` will get called again).

Reply:
Okay, indeed, seems there is no issue if we don't call `applyContent(if (something) content1 else content2)`.

> Especially since nothing terrible

If we would call the statement I mentioned, then we have a memory leak - `setContent {` captures lambda from the first recomposition.

Reply:
Ok, so there is a need to `rememberUpdatedState` the content here. Without it, nothing will happen when `content` changes. A new (non-composable) lambda will get created and passed to `DisposableEffect`, but because the key hasn't changed, it will be ignored.